File: go1_pulse/go1_pulse/report/employee_cost_allocation/employee_cost_allocation.py
# Copyright (c) 2023, Tridots Team and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
	return get_columns(filters), get_data(filters)

def get_columns(filters):
	columns = [
		{
			"fieldname": "employee",
			"label": "Employee",
			"fieldtype": "Link",
			"options": "Employee",
			"width": 150,
		},
		{
			"fieldname": "project",
			"label": "Project",
			"fieldtype": "Link",
			"options": "Project",
			"width": 150,
		}, 
		{
			"fieldname": "cost_center",
			"label": "Cost Center",
			"fieldtype": "Link",
			"options": "Cost Center",
			"width": 150,
		},
		{
			"fieldname": "total_working_days",
			"label": "Working Days",
			"fieldtype": "Float",
			"width": 150,
		},
		{
			"fieldname": "employee_worked_days",
			"label": "Worked Days",
			"fieldtype": "Float",
			"width": 150,
		},
		{
			"fieldname": "employee_worked_hours",
			"label": "Worked Hours",
			"fieldtype": "Float",
			"width": 150,
		},
		{
			"fieldname": "billing_amount",
			"label": "Billing Amount",
			"fieldtype": "Currency",
			"width": 150,
		},
		]
	
	return columns

def get_data(filters):
	return frappe.get_all("Timesheet Aggregator Detail", {"parent": filters.get("timesheet_aggregator")},
      	['employee', 'project', 'total_working_days', 'employee_worked_days', 'employee_worked_hours', 'billing_amount', 'cost_center']
        )

File: go1_pulse/go1_pulse/report/employee_cost_allocation/employee_cost_allocation.js
// Copyright (c) 2023, Tridots Team and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Employee Cost Allocation"] = {
	"filters": [
		{
			"fieldname": "timesheet_aggregator",
			"label": __("Timesheet Aggregator"),
			"fieldtype": "Link",
			"options": "Timesheet Aggregator",
			"reqd": 1,
			"read_only": 1
		},
	],

	onload: function (report) {
		report.page.add_inner_button(__("Make Journal"), function () {
			let dialog = new frappe.ui.Dialog({
				title: __("Select Account"),
				fields: [{
						"label": __("Account"),
						"fieldname": "account",
						"fieldtype": "Link",
						"options": "Account",
						"reqd": 1,
					},
					{
						"label": __("Posting Date"),
						"fieldname": "posting_date",
						"fieldtype": "Date",
						"reqd": 1,
					}],

				primary_action: function(){
					var args = dialog.get_values();
					frappe.call({
						method: "go1_pulse.go1_pulse.doctype.employee_salary_importer.employee_salary_importer.make_journal_entry",
						args: {
							"datas": report.data.slice(0, -1),
							"args": args,
							"ta_name": frappe.query_report.filters[0].value
						},
						callback: function (res) {
							
							if (res.message) {
								frappe.model.sync(res.message);
								frappe.set_route("Form", res.message.doctype, res.message.name);
							}
						}
					});
					dialog.hide();
				},
				
				primary_action_label: __("Create Journal Entry")
			});
			dialog.show();
		})		
	}
};

